thread-rcu/rcu.h

```diff
@@ -0,0 +1,216 @@
+#ifndef __RCU_H__
+#define __RCU_H__
+
+/* lock primitives from pthread and compiler primitives */
+
+#include <pthread.h>
+
+typedef pthread_mutex_t spinlock_t;
+
+#define SPINLOCK_INIT PTHREAD_MUTEX_INITIALIZER
+
+static __inline__ void spin_lock(spinlock_t *sp)
+{
+    int ret;
+
+    ret = pthread_mutex_lock(sp);
```

Comment:
Why not using [pthread_spin_lock](https://man7.org/linux/man-pages/man3/pthread_spin_lock.3.html)?

Reply:
Because this needs to add[ `-std=gnu99`](https://stackoverflow.com/questions/15673492/gcc-compile-fails-with-pthread-and-option-std-c99), but when I wrote this code I use `-std=c89`.

Reply:
Let's enforce C99.

Reply:
Ok.
Now, I change to `-std=c99`, so preserve the `pthread_mutex_t`.